src/phantom_token.c

```diff
@@ -265,63 +322,79 @@ static ngx_int_t handler(ngx_http_request_t *request)
             if (module_context->status == NGX_HTTP_OK)
             {
                 // Introspection was successful. Replace the incoming Authorization header with one that has the JWT.
-                request->headers_in.authorization->value.len = module_context->jwt.len;
-                request->headers_in.authorization->value.data = module_context->jwt.data;
+                if (headers_more_set_header_in(request, AUTHORIZATION_HEADER_NAME, module_context->jwt, &request->headers_in.authorization) != NGX_OK)
```

Comment:
The main changes are that all calls to update headers_in now use the utility module, but otherwise logic is the same.